spec/scenarios/pay_in_advance_charges_spec.rb

```diff
@@ -374,6 +374,268 @@
         expect(fee.amount_cents).to eq(400)
       end
     end
+
+    context 'when there is matching group' do
+      let(:transaction_id) { SecureRandom.uuid + 'test' }
+      let(:parent_group_id) { nil }
+      let(:group) do
+        create(:group, billable_metric:, key: 'region', value: 'europe', parent_group_id:)
+      end
+
+      it 'creates a pay_in_advance fee' do
+        ### 24 january: Create subscription.
+        jan24 = DateTime.new(2023, 1, 24)
+
+        travel_to(jan24) do
+          create_subscription(
+            {
+              external_customer_id: customer.external_id,
+              external_id: customer.external_id,
+              plan_code: plan.code,
+            },
+          )
+        end
+
+        charge = create(
+          :standard_charge,
+          :pay_in_advance,
+          invoiceable: true,
+          plan:,
+          billable_metric:,
+          group_properties: [
+            build(
+              :group_property,
+              group:,
+              values: {
+                amount: '20',
+                amount_currency: 'EUR',
+              },
+            ),
+          ],
+        )
+
+        subscription = customer.subscriptions.first
+
+        ### 15 february: Send an event.
+        feb15 = DateTime.new(2023, 2, 15)
+
+        travel_to(feb15) do
+          create_event(
+            {
+              code: billable_metric.code,
+              transaction_id:,
+              external_customer_id: customer.external_id,
+              properties: { amount: '10', region: 'europe' },
+            },
+          )
+
+          expect(subscription.reload.fees.count).to eq(1)
+          expect(Event.find_by(transaction_id:).metadata['current_aggregation']).to eq('10')
+
+          fee = subscription.fees.first
+
+          expect(fee.invoice_id).not_to be_nil
+          expect(fee.charge_id).to eq(charge.id)
+          expect(fee.pay_in_advance).to eq(true)
+          expect(fee.units).to eq(10)
+          expect(fee.events_count).to eq(1)
+          expect(fee.amount_cents).to eq(20000)
+        end
+      end
+    end
+
+    context 'when there is no matching group' do
+      let(:transaction_id) { SecureRandom.uuid + 'test' }
```

Comment:
Style/StringConcatenation: Prefer string interpolation to string concatenation. (https://rubystyle.guide#string-interpolation)